src/python/pants/help/help_printer.py

```diff
@@ -53,23 +55,13 @@ def print_hint() -> None:
         elif isinstance(self._help_request, AllHelp):
             self._print_all_help()
         elif isinstance(self._help_request, ThingHelp):
-            self._print_thing_help()
+            return self._print_thing_help()
```

Comment:
At this point, it would probably be clearer for all branches of the `if/else` to `return` explicitly, rather than to fall through.